src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -53,12 +56,360 @@ uint32_t SystemRequest::index = 0;
+                                 << languages_array_size << "]>["
+                                 << max_languages_array_size_ << "]");
+      result = false;
+    }
+    // Every language has ttsname string and vrsynonyms array
+    for (size_t idx = 0; idx < languages_array_size; ++idx) {
+      const smart_objects::SmartObject& language = languages.getElement(idx);
+      std::set<std::string> keys = language.enumerate();
+      for (std::set<std::string>::const_iterator iter = keys.begin();
+           iter != keys.end();
+           ++iter) {
+        // Verify default language defined
+        if (!(*iter).compare(json::default_)) {
+          default_language_found = true;
+        }
+        // Add set for synonyms' duplicates validation
+        if (synonyms_map_.find(*iter) == synonyms_map_.end())
+          synonyms_map_[*iter] = SynonymsSetType();
+        // ttsName verification
+        if (!language[*iter].keyExists(json::ttsName)) {
+          LOG4CXX_WARN(logger_,
+                       error_prefix_
+                           << "\"languages\".\"\ttsName\" doesn't exist");
+          result = false;
+        } else {
+          const smart_objects::SmartObject& ttsNameObject =
+              language[*iter][json::ttsName];
+          // ttsName is string
+          if (smart_objects::SmartType_String == ttsNameObject.getType()) {
+            const std::string ttsName =
+                language[*iter][json::ttsName].asString();
+            if (ttsName.length() > max_tts_name_length_) {
+              LOG4CXX_WARN(logger_,
+                           error_prefix_
+                               << "ttsName string exceeds max length ["
+                               << ttsName.length() << "]>["
+                               << max_tts_name_length_ << "]");
+              result = false;
+            }
+          } else {
+            LOG4CXX_WARN(logger_,
+                         error_prefix_ << "ttsName is not the string type.");
+            result = false;
+          }
+        }
+
+        if (!language[*iter].keyExists(json::vrSynonyms)) {
+          LOG4CXX_WARN(logger_,
+                       error_prefix_
+                           << "\"languages\".\"\vrSynonyms\" doesn't exist");
+          result = false;
+        } else {
+          const smart_objects::SmartArray* syn_array =
+              language[*iter][json::vrSynonyms].asArray();
+          if (NULL == syn_array) {
+            LOG4CXX_WARN(logger_, error_prefix_ << "vrSynonyms is not array.");
+            result = false;
+          } else {
+            const size_t syn_size = syn_array->size();
+            if (syn_size < min_vr_array_size_) {
+              LOG4CXX_WARN(logger_,
+                           error_prefix_ << "vrSynomyms array has [" << syn_size
+                                         << "] size < allowed min size ["
+                                         << min_vr_array_size_ << "]");
+              result = false;
```

Comment:
@okoreniak, I still don't get the point in this dcheck. Why there are no similar dchecs in other places?